### PR TITLE
Add docker cache directory

### PR DESCRIPTION
**Pull request by or-shachar**
These are the changes I made to make puller use symlink.

Note that we have 3 alternative:
- symlink_to_cache
- copy_from_cache
- hardlink_to_cache

But currently it's hardcoded to symlink. I think it should be configurable before opening a PR to the mother repo.

### CODE REVIEW

File: client/v2_2/save_.py [hunk 1]
```diff
@@ -173,23 +175,43 @@ def write_file(name, accessor,
...
+    os.symlink(source, dest)
+
+
+def hardlink_to_cache(source, dest):
+    # unlink first to remove "old" layers if needed, e.g., image A latest has layers 1, 2 and 3
+    # after a while it has layers 1, 2 and 3'. Since in both cases the layers are named 001, 002 and 003
+    # Unlinking promises the correct layers are linked in the image directory
+    if os.path.exists(dest):
+      os.unlink(dest)
+    os.symlink(source, dest)
```
Comment (anchlovi) [inline client/v2_2/save_.py]: I'm confused the method name is `hardlink...` but you are using `os.symlink`

**Review by anchlovi — COMMENTED** [review]
One minor issue - the next thing I don't understand is how do you use the three options? I don't see and additional parameter

File: client/v2_2/save_.py (hunk 1 above)
Comment (or-shachar) [reply client/v2_2/save_.py]: You're right!

Comment (or-shachar) [reply client/v2_2/save_.py]:
```suggestion
    os.link(source, dest)
```

File: client/v2_2/save_.py [hunk 2]
```diff
@@ -173,23 +175,43 @@ def write_file(name, accessor,
...
+    if os.path.exists(dest):
+      if not files_are_equal(source, dest):
+          os.remove(dest)
+          copy_file(source, dest)
+    else:
+        copy_file(source, dest)
+
+
+  def symlink_to_cache(source, dest):
     # unlink first to remove "old" layers if needed, e.g., image A latest has layers 1, 2 and 3
```
Comment (anchlovi) [inline client/v2_2/save_.py:194]: one last thing - comment is no longer relevant

**Pull request closed without merging**